### PR TITLE
QDEV-1285 new custom fields api

### PR DESCRIPTION
**Pull request by abelharisov**

### CODE REVIEW

File: Taskfile [hunk 1]
```diff
@@ -37,6 +37,7 @@ function ui() {
 
 swagger-codegen() {
   docker run --rm -it \
+    -v "$(pwd)"/php.sdk.config.json:/specs/php.sdk.config.json \
```
Comment (kamilsk) [inline Taskfile:40]: I suggest creating a config dir, as examples/sdk. and potentially examples/sdk/config

File: paths/custom_field.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+get:
+  operationId: custom-field
```
Comment (kamilsk) [inline paths/custom_field.yaml:2]: get-custom-field?

File: paths/custom_field.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  tags: [ custom_fields ]
```
Comment (kamilsk) [inline paths/custom_field.yaml:3]: custom-fields / custom-field?

File: schemas/CustomFieldCreate.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      0 - case;
+      1 - run;
+      2 - defect;
+  type:
+    type: integer
+    minimum: 0
+    maximum: 9
+    description: |
+      Possible values:
+      0 - number;
```
Comment (kamilsk) [inline schemas/CustomFieldCreate.yaml:32]: 0...9 are array keys?

File: src.yaml [hunk 5]
```diff
@@ -52,11 +61,16 @@ components:
...
+    IdResponse: { $ref: './responses/IdResponse.yaml' }
     Response: { $ref: './responses/Response.yaml' }
     SearchResponse: { $ref: './responses/SearchResponse.yaml' }
 
   parameters:
     Code: { $ref: './parameters/project/Code.yaml' }
+    Id: { $ref: './parameters/Id.yaml' }
     Limit: { $ref: './parameters/pagination/Limit.yaml' }
     Offset: { $ref: './parameters/pagination/Offset.yaml' }
+
```
Comment (kamilsk) [inline src.yaml:76]: double new line?

**Review by kamilsk — COMMENTED** [review]
first look LGTM

File: Taskfile (hunk 1 above)
Comment (abelharisov) [reply Taskfile]: I was committed by accident =)
Now this config is located with qase sources

File: paths/custom_field.yaml (hunk 2 above)
Comment (abelharisov) [reply paths/custom_field.yaml]: renamed

File: paths/custom_field.yaml (hunk 3 above)
Comment (abelharisov) [reply paths/custom_field.yaml:3]: custom_fields is a tag related to all custom fields api

File: schemas/CustomFieldCreate.yaml (hunk 4 above)
Comment (abelharisov) [reply schemas/CustomFieldCreate.yaml:32]: They are ids of custom fields types

File: src.yaml (hunk 5 above)
Comment (abelharisov) [reply src.yaml]: fixed

File: examples/custom_field.http [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# operationId: get-custom-field
```
Comment (kamilsk) [inline examples/custom_field.http:1]: please use one file for the entity: custom_fields.http

File: examples/custom_fields.http [hunk 7]
```diff
@@ -0,0 +1,18 @@
+# operationId: custom-fields
```
Comment (kamilsk) [inline examples/custom_fields.http:1]: please add verb as a prefix

File: paths/custom_field.yaml (hunk 3 above)
Comment (kamilsk) [reply paths/custom_field.yaml:3]: I suggest avoiding `_` and use `-` as a separator.

File: paths/custom_fields.yaml [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+              - checkbox
+              - selectbox
+              - radio
+              - multiselect
+              - url
+              - user
+              - datetime
+  responses:
+    200:
+      description: Custom Fields list.
```
Comment (kamilsk) [inline paths/custom_fields.yaml:37]: Custom Field list, the plural is already in the `list`.

File: schemas/CustomFieldCreate.yaml [hunk 9]
```diff
@@ -0,0 +1,64 @@
+type: object
```
Comment (kamilsk) [inline schemas/CustomFieldCreate.yaml:1]: discussion: what do you think about the next format `<EntityType>.<operation>.yaml`

- CustomField.create.yaml
- CustomField.update.yaml
- but: CustomFieldValue.yaml

File: schemas/responses/CustomFieldResponse.yaml [hunk 10]
```diff
@@ -0,0 +1,6 @@
+allOf:
```
Comment (kamilsk) [inline schemas/responses/CustomFieldResponse.yaml:1]: I suggest avoiding duplication in a path: `/resposes/` and `*Response.yaml`

just simplify it to the `schemas/responses/CustomField.yaml` or `schemas/responses/CustomField.success.yaml`

File: schemas/responses/CustomFieldsResponse.yaml [hunk 11]
```diff
@@ -0,0 +1,17 @@
+allOf:
```
Comment (kamilsk) [inline schemas/responses/CustomFieldsResponse.yaml:1]: the same suggestion as above

**Review by kamilsk — CHANGES_REQUESTED** [review]
I like it, but let's do little improvements.

**Comment by kamilsk** [comment]
please fix it in the next pr

**Pull request merged**